PennMobile/Events/Views/PennEventsViewerView.swift

```diff
@@ -0,0 +1,234 @@
+//
+//  PennEventsViewerView.swift
+//  PennMobile
+//
+//  Created by Jacky on 3/10/24.
+//  Copyright © 2024 PennLabs. All rights reserved.
+//
+
+import SwiftUI
+import MapKit
+import SafariServices
+import MessageUI
+
+struct PennEventsViewerView: View {
+    var event: PennEventViewModel
+    
+    @Environment(\.presentationMode) var presentationMode
+    
+    // boolean flags for the webView and link button
+    @State private var showingWebView = false
+    @State private var showAlert = false
+    
+    // for email button
+    @State private var showingMailComposer = false
+    
+    // coordinates for the map annotation
+    @State private var eventCoordinate: CLLocationCoordinate2D?
+    
+    // default to philly
+    @State private var region = MKCoordinateRegion(
+        center: CLLocationCoordinate2D(latitude: 39.9522, longitude: -75.1932),
+        span: MKCoordinateSpan(latitudeDelta: 0.0020, longitudeDelta: 0.0020)
```

Comment:
Probably use `PennCoordinate.shared.getDefaultRegion(at: )` instead for the default value for consistency. Look into this file and see if there's anywhere else where these helper methods might be useful.

Reply:
Move to same shared location file